source/cppfront.cpp

```diff
@@ -1928,6 +1933,18 @@ class cppfront
         }
         suppress_move_from_last_use = false;
 
+        if (args) {
+            // if after printing core expression args is defined
+            // it means that the chaining started by function call
+            // we need to print its arguments
+            suffix.emplace_back(")", n.position());
+            for (auto&& e: args.value()) {
+                suffix.push_back(e);
+            }
+            suffix.emplace_back("(", n.position());
```

Comment:
These two `suffix.` lines should also use the positions:

```
            suffix.emplace_back(")", args_close_pos);
            ...
            suffix.emplace_back("(", args_open_pos);
```

That should do it!